pkg/action/action_handler.go

```diff
@@ -265,17 +165,61 @@ func getActionTypeFromActionItemDTO(actionItem *proto.ActionItemDTO) (turboactio
 	return actionType, err
 }
 
-// Send action response to Turbonomic server.
-func (handler *ActionHandler) sendActionResult(state proto.ActionResponseState, progress int32, description string) {
-	// 1. build response
-	response := &proto.ActionResponse{
+func (h *ActionHandler) goodResult() *proto.ActionResult {
+
+	state := proto.ActionResponseState_SUCCEEDED
+	progress := int32(100)
+	msg := "Success"
+
+	res := &proto.ActionResponse{
 		ActionResponseState: &state,
 		Progress:            &progress,
-		ResponseDescription: &description,
+		ResponseDescription: &msg,
 	}
-	// 2. built action result.
-	result := &proto.ActionResult{
-		Response: response,
+
+	return &proto.ActionResult{
+		Response: res,
 	}
-	handler.resultChan <- result
+}
+
+func (h *ActionHandler) failedResult(msg string) *proto.ActionResult {
+
+	state := proto.ActionResponseState_FAILED
+	progress := int32(0)
```

Comment:
It doesn't make sense to explicitly set the progress to 0 if the action has reached to final state (regardless of succeed or fail)

Reply:
so, set it to 100, or -1?

Reply:
We can just remove the line.